webapp/components/Form.tsx

```diff
@@ -57,6 +66,42 @@ const Form = (): JSX.Element => {
     </div>
   );
 
+  const fetchUserData = async () => {
+    let token = ''
+    try {
+      const response = await fetch('http://localhost:8000/token/', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      body: JSON.stringify(loginBody)
+    });
+      const json = await response.json();
+      token = json.access
+    }
+    catch (err) {
+      console.log(err)
+    }
+
+    try {
+      const response = await fetch('http://localhost:8000/users/profile/', {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+          'Authorization': `Bearer ${token}`
+        }
+      })
+      const json = await response.json();
+      setName(`${json.first_name} ${json.last_name}`)
+      setMailfrom(json.email)
+      setAccountNumber(json.bank_account_number)
+
+    }
+    catch (err) {
```

Comment:
From [fetch-docs](https://developer.mozilla.org/en-US/docs/Web/API/fetch)

> A [fetch()](https://developer.mozilla.org/en-US/docs/Web/API/fetch) promise only rejects when a network error is encountered (which is usually when there's a permissions issue or similar). A [fetch()](https://developer.mozilla.org/en-US/docs/Web/API/fetch) promise does not reject on HTTP errors (404, etc.). Instead, a then() handler must check the [Response.ok](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok) and/or [Response.status](https://developer.mozilla.org/en-US/docs/Web/API/Response/status) properties.


Thus triggering 401 will not go to the catch-clause and you set the name, mail and account number even if there is none to set.

Add a check if the response-object is ok or not like they have done further down in this file or as explained in [the fech-docs](https://developer.mozilla.org/en-US/docs/Web/API/fetch#examples).

Reply:
> If you are to do this, you need to do some good error-handling if something fails on the retrieval of groups. If the user decides _not_ to use the login-to-retrieve-data-functionality. You must display all groups. [Mui has some nifty autocomplete-components ](https://mui.com/components/autocomplete/) which would make retrieving all groups and searching in them fast anyway. Retrieving only the groups you are member of is a nice feature, though ⭐

Would you consider it a bad practice to change the component to a MUI Dropdown if the user has memberships in one group or more? E.g. something like this:

```javascript
groups = fetchGroups()
{ groups.length >= 1 ? <Dropdown prop1=".." prop2=".." /> : <Input prop1=".." prop2=".."/>
```
Or maybe it is just simpler to fetch all groups either way and use the same dropdown whether the user fetches data or not. 

> I see nothing wrong with this one other that it will require one request every time you change the group-dropdown which again requires some proper error-handling. Nice feature ⭐

What if we have a `[groups, setGroups] = useState([])` which contains all the groups? Here we will also have the `gsuite_prefix` which we can use for the cashier email, i.e. 
`setMailto('${selectedGroup.gsuite_prefix}-kasserer@ntnui.no')`

Reply:
> change the component to a MUI Dropdown if the user has memberships in one group or more

If the user has memberships in no groups it would be a normal input-field, not a dropdown? Perhaps it would be better to just give the user a dropdown of all groups in that case? Or am I misunderstanding?

It might better to just fetch all groups. It will be an improvement from before either way.

> What if we have a [groups, setGroups] = useState([]) which contains all the groups?

That would be the easiest solution I believe 🧷

Reply:
Nice! I think we're onto something good 🤓 Will fix 💪